RIOT-OS/RIOT
# pkg/tinyusb: add tinyUSB as package

### Contribution description

This PR adds [tinyUSB](https://github.com/hathach/tinyusb) open-source cross-platform USB Host/Device stack for embedded systems as package.

The initial intention was to add USB support for ESP32-S2 and ESP32-S3 since the tinyUSB stack includes a hardware driver for these ESP32x SoCs. The ESP-IDF also uses tinyUSB for USB support, but integrates tinyUSB. Instead of using the tinyUSB from ESP-IDF, this PR adds tinyUSB as a package to be able to use it for other platforms like STM32 and nRF. STM32 support is also provided by this PR.

### Testing procedure
 
Compile and flash the test application `tests/pkg_tinyusb_cdc_msc`.
```
BOARD=esp32s3-devkit make -j8 -C tests/pkg_tinyusb_cdc_msc flash
```
It should be possible to use any STM32 board that also supports feature `periph_usbdev`

This application uses the tinyUSB device stack to emulate a mass storage device (MSC) with a communication interface (CDC).

Once the application is flashed, the device should be mounted when it is
connected to a host. That is,

- the mass storage interface is mounted as volume `TinyUSB MSC` in the operating system and
- the communication interface is mounted as a serial device, for example as `/dev/ttyACM0` on Linux.

It should then be possible

1. to read from and write to the mass storage device with the usual file operations of the operating system
2. to connect to the serial interface of the device with a terminal program, e.g.
   ```
   python -m serial.tools.miniterm /dev/ttyACM0 115200
   ```
   and get the entered characters sent back.

The test application uses LED0, if present, to indicate the status of the device by blinking at different frequencies.

### Issues/PRs references
